addOns/frontendscanner/src/main/java/org/zaproxy/zap/extension/frontendscanner/FrontEndScannerAPI.java

```diff
@@ -76,14 +78,20 @@ public String handleCallBack(HttpMessage msg) throws ApiException {
                 throw new ApiException(ApiException.Type.MISSING_PARAMETER, "action");
             }
 
-            Map<String, String> parameters = parameterParser.getParams(msg, HtmlParameter.Type.url);
+            List<NameValuePair> parameters =
+                    parameterParser.getParameters(msg, HtmlParameter.Type.url);
 
-            String action = parameters.get("action");
+            String action =
+                    parameters.stream()
+                            .filter(a -> a.getName().equals("action"))
+                            .findFirst()
+                            .get()
```

Comment:
This will throw a `NoSuchElementException` when the parameter is not present (same comment for the other changes).

Reply:
Thanks for catching that. I’ll hop into IRC later to discuss.